osgp/platform/osgp-adapter-kafka-distributionautomation/src/test/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/LowVoltageMessageFactory.java

```diff
@@ -0,0 +1,115 @@
+import com.alliander.data.scadameasurementpublishedevent.UnitMultiplier;
+import com.alliander.data.scadameasurementpublishedevent.UnitSymbol;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LowVoltageMessageFactory extends MessageFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(LowVoltageMessageFactory.class);
+
+  public enum Version {
+    VERSION_1,
+    VERSION_2;
+  }
+
+  public static List<Analog> expectedMeasurements(final Version version) {
+    if (Version.VERSION_1 == version) {
+      return expectedMeasurementsVersion1();
+    } else if (Version.VERSION_2 == version) {
+      return expectedMeasurementsVersion2();
+    } else {
+      LOGGER.error("Not given the an expected version. Given version: {}", version);
+      return null;
+    }
+  }
+
+  public static List<Analog> expectedMetaMeasurements() {
+    final List<Analog> measurements = new ArrayList<>();
+    measurements.add(createAnalog("Frequency", 49.98f, UnitSymbol.Hz));
+    measurements.add(createAnalog("Temperature", 12.0f, UnitSymbol.degC));
+    measurements.add(createAnalog("IRMS-N", 0.11f, UnitSymbol.A));
+    return measurements;
+  }
+
+  private static List<Analog> expectedMeasurementsVersion1() {
+    final List<Analog> measurements = new ArrayList<>();
+    measurements.add(createAnalog("U-L1", 0.1f, UnitSymbol.V));
+    measurements.add(createAnalog("U-L2", 0.2f, UnitSymbol.V));
+    measurements.add(createAnalog("U-L3", 0.3f, UnitSymbol.V));
+    generalExpectedMeasurements(measurements, 0.4);
+    return measurements;
+  }
+
+  private static List<Analog> expectedMeasurementsVersion2() {
+    final List<Analog> measurements = new ArrayList<>();
+    measurements.add(createAnalog("U-avg", 0.1f, UnitSymbol.V));
+    generalExpectedMeasurements(measurements, 0.2);
+    double value = measurements.size() * 0.1;
+    measurements.add(createAnalog("IrmsN", (float) (value = value + 0.1), UnitSymbol.A));
+    measurements.add(createAnalog("Pp", (float) (value = value + 0.1), UnitSymbol.none));
+    measurements.add(createAnalog("Pm", (float) (value = value + 0.1), UnitSymbol.none));
+    measurements.add(createAnalog("Qp", (float) (value = value + 0.1), UnitSymbol.none));
+    measurements.add(createAnalog("Qm", (float) (value = value + 0.1), UnitSymbol.none));
+    measurements.add(createAnalog("U-L1", (float) (value = value + 0.1), UnitSymbol.V));
+    measurements.add(createAnalog("U-L2", (float) (value = value + 0.1), UnitSymbol.V));
+    measurements.add(createAnalog("U-L3", (float) (value = value + 0.1), UnitSymbol.V));
+    measurements.add(createAnalog("Temp", (float) (value = value + 0.1), UnitSymbol.C));
+    measurements.add(createAnalog("F", (float) (value = value + 0.1), UnitSymbol.Hz));
+    return measurements;
+  }
+
+  private static void generalExpectedMeasurements(final List<Analog> measurements, double d) {
```

Comment:
Consider the following improvements:
* Create a new list and return this list, instead of working on a List passed as parameter directly 
* Use a float parameter instead of casting each value to float
* Rename the float parameter to make the meaning more explicit

```suggestion
  private static List<Analog> createCommonExpectedMeasurements(final float startValue) {
    List<Analog> measurement = new ArrayList<>();
    float value = startValue;
```